### PR TITLE
An error message is added if user enters wrong credentials while login and the UI of /accounts/confirm-email page is fixed

### LINKED ISSUE #4056: Show an error message on incorrect login credentials and fix verify email page UI.
The login functionality does not prompt user with an error message on incorrect credentials. Also the verify email page UI is broken.

Also please allow username and email both with password for sign in

https://github.com/user-attachments/assets/74fb5c4b-4000-467e-a934-9a99ad432566

### PR DESCRIPTION
**Pull request by swaparup36**
It resolves #4056 

The error message for the wrong credentials while login -
Screenshot 2025-03-27 200429

The UI of /accounts/confirm-email/ page -
Screenshot 2025-03-27 221557

## Summary by CodeRabbit

* **New Features**
  * Login form now labeled "Email" (translatable), accepts email input, displays non-field errors and color-coded status messages; added "Already Verified" indicator, spam-folder reminders, helpful tips, and changed action wording to "Resend Verification Email".
  * Verification page redesigned into a two-panel "Check Your Email" layout with clearer Login/Sign Up CTAs and translated metadata.
* **Chores**
  * Site authentication switched to email-based login (username not required).
* **Tests**
  * Test suite updated to use email-based authentication and ensure email-verified preconditions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by DonnieBLT — DISMISSED** [review]
please change to email login, looks great otherwise, nice work

File: blt/settings.py [hunk 1]
```diff
@@ -344,6 +344,7 @@
 
 ACCOUNT_EMAIL_REQUIRED = True
 ACCOUNT_USERNAME_REQUIRED = True
+ACCOUNT_AUTHENTICATION_METHOD = "username_email"
```
[the remark this answers is not in the document]
Comment (swaparup36) [reply blt/settings.py:347]: Okay okay, I am doing that

[the remark this answers is not in the document]
Comment (swaparup36) [reply blt/settings.py]: I changed this and modified the tests such that matches the new login

**Comment by swaparup36** [comment]
Sure

**Review by sidd190 — COMMENTED** [review]
LGTM, just consider the comment and it's all good as far as I can see. Thanks for the PR!

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
please allow username and email both with password for sign in

**Comment by swaparup36** [comment]
@coderabbitai can you give me the fix for the test failing error

**Comment by swaparup36** [comment]
@coderabbitai Can you help me to fix the test error now

**Review by sidd190 — APPROVED** [review]
LGTM overall. Can you just consider if we need to do any of these and look over them quickly once?


* [ ] Add a conditional on the confirm-email page to show an appropriate message when the confirmation link is invalid or expired (`{% if confirmation %} ... {% else %} ... {% endif %}`).
* [ ] Update the remaining test that still posts `username` instead of `login`/`email` after signup.
* [ ] Verify both login methods (email and username) pass consistently in tests.

**Comment by swaparup36** [comment]
@sidd190 Sure, all of these are already done. And for the remaining tests that still posts username instead of login/email after signup is fine because both of the methods will work for login.

**Review by Pritz395 — CHANGES_REQUESTED** [review]
Hey, great work overall! Just flagging a few things that need attention:

1. Hardcoded URLs
- FYI - I already fixed these hardcoded URLs in PR #5132 using Django URL template tags. Since those files were changed outside this PR's scope, consider leaving them as-is rather than re-doing the fixes here.

2. Missing Features on Verification Page
The new verification page lost some functionality from the original:
- "Resend verification email" button
- SEO metadata (page titles, descriptions, Open Graph tags)
- Spam folder reminder for users
- Error handling for expired/invalid verification links
- These need to be restored.

3. Technical Issues
- Login error handling: The form currently shows "Invalid email or password" for all errors (including system errors like CSRF failures). This should only display for actual authentication failures.
- Dark mode styling: Some text styling was removed, breaking readability in dark mode.

4. Code Duplication in Tests
- Lines 162-180 and 204-222 have duplicate user setup code. Consider extracting this into a helper function (something like _ensure_bugbug_verified()) to keep things DRY.

Optional : 5. Test Coverage Gaps
Could you add tests for:
- Unverified users attempting to log in
- Different error scenarios (CSRF, rate limiting, etc.)
- Edge cases

**Comment by Pritz395** [comment]
@coderabbitai verification_sent.html — missing features:
Lines 1-6: No {% block title %}, {% block description %}, {% block keywords %}, {% block og_title %}, {% block og_description %} — confirmed missing
Grep for "resend|account_email": no matches — confirmed missing
No conditional blocks for expired/invalid links — confirmed missing
login.html line 50 — dark mode:
Line 50: <h3 class="text-2xl font-semibold text-gray-700"> — no dark:text-gray-300 — confirmed
login.html line 51 — error handling:
Line 51: {% if form.errors %} — not form.non_field_errors — confirmed
Test duplication:
test_post_bug_full_url() (lines 196-212) and test_post_bug_domain_url() (lines 246-262) have identical user setup code — confirmed duplicate. Have these been addressed?

**Review by Pritz395 — APPROVED** [review]
Niceee work addressing the feedback! LGTM.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Login should be email only

**Comment by swaparup36** [comment]
@coderabbitai Can you give me the fix for the test error

File: blt/settings.py [hunk 2]
```diff
@@ -342,7 +342,7 @@
     DATABASES["default"] = dj_database_url.config(conn_max_age=0, ssl_require=False)
 
 ACCOUNT_EMAIL_REQUIRED = True
-ACCOUNT_USERNAME_REQUIRED = True
+ACCOUNT_AUTHENTICATION_METHOD = "email"
```
Comment (DonnieBLT) [inline blt/settings.py]: don't we want to require a username?

**Comment by DonnieBLT** [comment]
this seems like it fixes too much - please only focus on fixing the issue
